torch_xla/csrc/elementwise.cpp

```diff
@@ -66,8 +66,16 @@ xla::XlaOp BuildThreshold(xla::XlaOp input, xla::XlaOp output,
 
 xla::XlaOp BuildRelu(xla::XlaOp input) {
   const xla::Shape& input_shape = ShapeHelper::ShapeOfXlaOp(input);
+#ifndef EXPERIMENTAL_XLA_UNBOUNDED_DYNAMISM
   return xla::Max(input, XlaHelpers::ScalarValue<float>(
                              0, input_shape.element_type(), input.builder()));
+#else
+  xla::XlaOp scalar = XlaHelpers::ScalarValue<float>(
+      0, input_shape.element_type(), input.builder());
+  auto promoted = XlaHelpers::Promote(input, scalar);
+
+  return xla::Max(promoted.first, promoted.second);
```

Comment:
can we add a comment to explain about this change?

Reply:
Sure, the xla::Max doesn't do the implicit broadcasting for unbounded dynamism, so we need to promote in torch_xla, will leave a comment there.

Reply:
Comment added